presto-main/src/test/java/com/facebook/presto/type/TestJsonOperators.java

```diff
@@ -197,4 +204,16 @@ public void testCastFromVarchar()
         assertFunction("cast('abc' as JSON)", JSON, "\"abc\"");
         assertFunction("cast('\"a\":2' as JSON)", JSON, "\"\\\"a\\\":2\"");
     }
+
+    @Test
+    public void testCastFromNestedDecimalTypes() {
+        assertFunction("cast(ARRAY[DECIMAL '1234567890.1234567890'] as JSON)", JSON, "[1234567890.1234567890]");
+        assertFunction("cast(ARRAY[ARRAY[DECIMAL '1234567890.1234567890']] as JSON)", JSON, "[[1234567890.1234567890]]");
+        assertFunction("cast(MAP(ARRAY['abc'], ARRAY[DECIMAL '1234567890.1234567890']) as JSON)", JSON,
+                "{\"abc\":1234567890.1234567890}");
+        assertFunction("cast(MAP(ARRAY['abc'], ARRAY[MAP(ARRAY['def'], ARRAY[DECIMAL '1234567890.1234567890'])]) as JSON)", JSON,
+                "{\"abc\":{\"def\":1234567890.1234567890}}");
+        assertFunction("cast(ROW(DECIMAL '1234567890.1234567890') AS JSON)", JSON, "[1234567890.1234567890]");
+        assertFunction("cast(ROW(ROW(DECIMAL '1234567890.1234567890')) AS JSON)", JSON, "[[1234567890.1234567890]]");
+    }
```

Comment:
Ouch, fair. We don't have comparable arrays implemented in serializer.
Please disregard my comment than.